### DISCUSSION TITLE
Add $url file name template tag

### DISCUSSION THREAD

**Discussion by bgeery** [Tyrrrz/YoutubeDownloader#244] [General]
Many times I'm looking for the URL of a deleted video from a playlist.  If we had a $URL tag I could search my downloaded videos for the matching url tag and I'd know the title of the deleted video.  All the tag needs to represent is the part of the URL after the v=:
`https://www.youtube.com/watch?v=5Fl--PVAf-U`
Thanks.

**Comment by bgeery** [comment]
Ah, I have another reason of needing this feature-- avoiding filename collisions.

I downloaded my 1500 song playlist and was curious why my downloads were a dozen songs short.  Turns out, I had 13 pairs of songs that match in both $title and $author tags, YoutubeDownloader was just skipping the second matches altogether.

I redownloaded the playlist adding the $num tag, then stripped that from the filenames using other software, thus revealing the conflicting filenames.

BTW, the songs are mostly different, even with the matching filenames.  It's mostly different version of the same song by the same artist but poorly titled on YouTube.  But there are a few duplicated "topic" versions, I'm going to have to purge.

**Reply by Tyrrrz** [reply]
I created #245

**Comment by bgeery** [comment]
Hey, thanks Tyrrrz.  Great software!
